firmware/Core/Src/littlefs/littlefs_helper.c

```diff
@@ -108,126 +108,156 @@ int8_t LFS_mount()
 {
-    if (!LFS_is_lfs_mounted)
-    {
-        DEBUG_uart_print_str("LittleFS not mounted.\n");
+    // Check if LFS is mounted
+    if (!LFS_is_lfs_mounted) {
+        LOG_message(LOG_SYSTEM_LFS, LOG_SEVERITY_WARNING, LOG_all_sinks_except(LOG_SINK_FILE), "LittleFS not mounted.");
         return 1;
     }
 
+    // Try to open the directory
     lfs_dir_t dir;
     int8_t open_dir_result = lfs_dir_open(&LFS_filesystem, &dir, root_directory);
     if (open_dir_result < 0)
     {
-        DEBUG_uart_print_str("Error opening a directory.\n");
+        LOG_message(LOG_SYSTEM_LFS, LOG_SEVERITY_CRITICAL, LOG_all_sinks_except(LOG_SINK_FILE), "Error opening directory: %s", root_directory);
         return open_dir_result;
     }
 
-    // result is positive on success, 0 at the end of directory, or  negative on failure.
+    if (count == 0) {
+        count = -1;
+    }
+
+    // result is positive on success, 0 at the end of directory, or negative on failure.
     int8_t read_dir_result = 1;
-    while (read_dir_result >= 0)
+    struct lfs_info info;
+    DEBUG_uart_print_str("Name \t bytes\n");
+    while (read_dir_result > 0)
     {
-        struct lfs_info info;
         read_dir_result = lfs_dir_read(&LFS_filesystem, &dir, &info);
 
-        DEBUG_uart_print_str(info.name);
-        DEBUG_uart_print_str(", ");
-        // TODO: The info struct contains information about directory contents
-    }
-    DEBUG_uart_print_str("\n");
+        if (read_dir_result < 0)
+        {
+            DEBUG_uart_print_str("Error reading directory contents.\n");
+            LOG_message(LOG_SYSTEM_LFS, LOG_SEVERITY_CRITICAL, LOG_all_sinks_except(LOG_SINK_FILE), "Error reading content from directory: %s", root_directory);
+            break;
+        }
 
-    if (read_dir_result < 0)
-    {
-        DEBUG_uart_print_str("Error reading directory contents.\n");
-        return read_dir_result;
-    }
+        if (offset > 0) {
+            offset--;
+            continue;
+        }
 
-    DEBUG_uart_print_str("Successfully Listed Directory Contents.\n");
+        if (count == 0) {
+            break;
+        } else {
+            count--;
+        }
+
+        DEBUG_uart_print_str(info.name);
+        if (info.type == LFS_TYPE_REG)
+        {
+            DEBUG_uart_print_str("\t");
+            DEBUG_uart_print_uint32(info.size);
+            DEBUG_uart_print_str(" bytes");
+        }
+        DEBUG_uart_print_str("\n");
+    }
 
     int8_t close_dir_result = lfs_dir_close(&LFS_filesystem, &dir);
     if (close_dir_result < 0)
     {
         DEBUG_uart_print_str("Error closing directory.\n");
+        LOG_message(LOG_SYSTEM_LFS, LOG_SEVERITY_CRITICAL, LOG_all_sinks_except(LOG_SINK_FILE), "Error closing directory: %s", root_directory);
         return close_dir_result;
     }
 
-    return 0;
+    if (read_dir_result < 0) {
+        return read_dir_result;
+    } else {
+        DEBUG_uart_print_str("Successfully Listed Directory Contents.\n");
```

Comment:
Please remove DEBUG_uart_print_str. Please try to find all instances in the new code you're adding here.